Git Formats/Git Log.sublime-syntax

```diff
@@ -13,13 +13,32 @@ contexts:
   main:
     # 1st header line
     # commit d9d9fb804f5d61c13ba2f8746af33a9f3c609075
-    - match: (?:(commit)\s+(\h{7,}))?\s*\n
-      scope: meta.header.git.commit markup.raw.block.git.log
+    - match: ^(commit)\s+(\h{7,})
       captures:
         1: keyword.other.commit.git.log
         2: constant.other.hash.git.log
-      embed: commit-header
-      escape: (?=^commit\s)
+      push:
+        - meta_scope: meta.header.git.commit markup.raw.block.git.log
+        - match: $\n?
+          scope: meta.header.git.commit markup.raw.block.git.log
```

Comment:
this `scope` shouldn't be needed due to the `meta_scope`, but ST doesn't apply it to the match which pops if it also does an `embed`, it seems

Reply:
Yes, all combined `pop-...` statements handle the match as it was a lookahead.

It is intended behavior and documentet.

Filed an [issue](https://github.com/sublimehq/sublime_text/issues/3783) about it as I found it inconsistent in situations like that.

But also found situations, it helps keeping things clean, so I don't have a very strong opinion about this behavior.

Basically it's more easy for us to add an extra line of `scope:` than to remove unwanted scopes.

This behaviour  just reduces the re-usability of such patterns as they are tight to the context they are used int.